packages/frontend/src/components/json_import.tsx

```diff
@@ -0,0 +1,109 @@
+import { createSignal } from "solid-js";
+import type { Document } from "../api";
+import "./json_import.css";
+
+interface JsonImportProps<T extends string> {
+    onImport: (data: Document<T>) => void;
+    validate?: (data: Document<T>) => boolean | string;
+}
+
+export const JsonImport = <T extends string>(props: JsonImportProps<T>) => {
+    const [error, setError] = createSignal<string | null>(null);
+    const [pasteValue, setPasteValue] = createSignal("");
```

Comment:
nit: textAreaValue (or similar) is likely a more appropriate name. This is not necessarily the value directly from the users clipboard.